### DISCUSSION TITLE
Is there a function for checking what the radio's doing?

### DISCUSSION THREAD

**Discussion by maalos** [jgromes/RadioLib#1358] [Q&A] [answered]
Is there a function for checking what the radio's doing? - Like receiving, listening/standby, transmitting, sleeping etc.

**Comment by HeadBoffin** [comment]
Depends on the radio ...

And as they only do what they've been told to do, perhaps a state variable in your own code will help.

Further comment would be dependent on knowing **WHICH RADIO** you are thinking of, because there's a few and well ....

**Reply by maalos** [reply]
sx126x/llcc68 in this case, the problem is I want to have a variable set to true while I am receiving until I stop receiving completely, for eg. 500ms - so 500ms after the last packet is received the variable will be set to false again.

**Reply by HeadBoffin** [reply]
There's a whole scheme in the radio for triggering an output that could be polled or triggered via an interrupt for you to explore that links to send, receive, CAD and more.

Are you using an RTOS or hoping to do something else whilst the radio is receiving?

**Reply by maalos** [reply]
I'm using FreeRTOS, I don't know how to set DIO actions

**Comment by maalos** [comment]
same for getFrequency, getSpreadFactor etc., is there a reason for why the library doesn't just provide those?

**Reply by StevenCellist** [reply, marked as answer]
See #622. But as mentioned by @HeadBoffin, you can just keep track of it in your code. If you set the frequency to 830 MHz, simply save the variable so that you know it. And why don't you `startReceive()`, wait for 500ms and put to `standby()`?

**Reply by HeadBoffin** [reply]
Or if you want to do something, start the receive, record the time/ticks, go off to do that other thing, come back to a delay that finishes off the receive window and then check in with the radio.

So many techniques, hard to suggest definitively without a heap more context.
